### PR TITLE
Implementing the "get" function on a "Dictionary"

### LINKED ISSUE #105: The "get" function is not implemented in the Dict.
in gpython
``` python
>>>> d = {'a' : 1}
>>>> d.get('a')
Traceback (most recent call last):
  File "test.py", line 13, in <module>
    FIXME line of source goes here
AttributeError: "'dict' has no attribute 'get'"
2019/09/29 16:24:04 AttributeError: "'dict' has no attribute 'get'"
```

in python
``` python
>>>> d = {'a' : 1}
>>>> d.get('a')
1
```

### PR DESCRIPTION
**Pull request by Sungmin-Joo**
Implement the "get" function and add unit test.
fixes #105

### CODE REVIEW

File: py/dict.go [hunk 1]
```diff
@@ -36,6 +36,29 @@ func init() {
 		}
 		return NewIterator(o), nil
 	}, 0, "items() -> list of D's (key, value) pairs, as 2-tuples")
+
+	StringDictType.Dict["get"] = MustNewMethod("get", func(self Object, args Tuple) (Object, error) {
+		var length = len(args)
+		if length == 0 {
```
Comment (sbinet) [inline py/dict.go]: I believe "Effective Go" recommands to use `switch` for these kind of `if-else-if` code paths:
```go
switch {
case length == 0:
   // ...
case length > 2:
   // ...
}
...
```

File: py/dict.go [hunk 2]
```diff
@@ -36,6 +36,29 @@ func init() {
...
+			return nil, ExceptionNewf(TypeError, "%s expected at least 1 arguments, got %d", "items()", length)
+		} else if length > 2 {
+			return nil, ExceptionNewf(TypeError, "%s get expected at most 2 arguments, got %d", "items()", length)
+		}
+		sMap := self.(StringDict)
+		if str, ok := args[0].(String); ok {
+			res, ok := sMap[string(str)]
+			if ok {
```
Comment (sbinet) [inline py/dict.go]: ditto:
```go
switch res, ok := sMap[string(str)]; ok {
case ok:
    return res, nil
case length == 1:
    return None, nil
default:
    return args[1], nil
}
```
```

Comment (Sungmin-Joo) [reply py/dict.go]: @sbinet 
``` go
switch res, ok := sMap[string(str)]; ok {
case ok:
    return res, nil
case length == 1:
    return None, nil
default:
    return args[1], nil
}
```
I applied as you advised, but it caused a "panic".
So I made some modifications.

**Comment by Sungmin-Joo** [comment]
@ncw 
Can I ask you to confirm my PR once again?

**Pull request merged**